R/splitModels.R

```diff
@@ -9,52 +20,47 @@ splitModels <- function ( qMatrix = NULL , person.groups = NULL , split = c ( "q
-				person.groups <- NULL
-				warning ( paste0 ( "splitModels: person.groups is not a data.frame and will be ignored." ) , call. = FALSE )
-		}
+#		if ( !is.null ( qMatrix ) & !is.data.frame ( qMatrix ) ) {
+#				qMatrix <- NULL
+#				warning ( paste0 ( "splitModels: qMatrix is not a data.frame and will be ignored." ) , call. = FALSE )
+#		}
+#		if ( !is.null ( person.groups ) & !is.data.frame ( person.groups ) ) {
+#				person.groups <- NULL
+#				warning ( paste0 ( "splitModels: person.groups is not a data.frame and will be ignored." ) , call. = FALSE )
+#		}
 		# wenn keine Spalten / Zeilen dann NULL
-		if ( !is.null ( qMatrix ) ) {
-				if ( nrow ( qMatrix ) %in% 0 | ncol ( qMatrix ) %in% 0 ) {
-						warning ( "splitModels: check qMatrix" , call. = FALSE )
-						qMatrix <- NULL
-				}
-		}
-		if ( !is.null ( person.groups ) ) {
-				if ( nrow ( person.groups ) %in% 0 | ncol ( person.groups ) %in% 0 ) {
-						warning ( "splitModels: check person.groups" , call. = FALSE )
-						person.groups <- NULL
-				}
-		}
+#		if ( !is.null ( qMatrix ) ) {
+#				if ( nrow ( qMatrix ) %in% 0 | ncol ( qMatrix ) %in% 0 ) {
+#						warning ( "splitModels: check qMatrix" , call. = FALSE )
+#						qMatrix <- NULL
+#				}
+#		}
+#		if ( !is.null ( person.groups ) ) {
+#				if ( nrow ( person.groups ) %in% 0 | ncol ( person.groups ) %in% 0 ) {
+#						warning ( "splitModels: check person.groups" , call. = FALSE )
+#						person.groups <- NULL
+#				}
+#		}
 		# Dimensionen in qMatrix duerfen nur 0/1 haben
-		if ( !is.null ( qMatrix ) ) {
-				if ( ncol ( qMatrix ) > 1 ) {
-						not01 <- ! sapply ( qMatrix[,-1,drop=FALSE] , function ( x ) all ( x %in% c(0,1) ) )
-						
-						if ( any ( not01 ) ) {
-								warning ( paste0 ( "splitModels: column(s) '" , paste ( names (not01)[not01] , collapse = "', '" ) , "' in qMatrix contain elements that are not 0 or 1; this/these column(s) are ignored" ) , call. = FALSE )
-								qMatrix <- qMatrix[,colnames(qMatrix)[!colnames(qMatrix) %in% names (not01)[not01]],drop=FALSE]
-						}
+		if ( test_data_frame(qMatrix, min.cols = 2) ) {
+		  not01 <- ! sapply ( qMatrix[,-1,drop=FALSE] , function ( x ) all ( x %in% c(0,1) ) )
+
+			if ( any ( not01 ) ) {
+			  warning ( paste0 ( "splitModels: column(s) '" , paste ( names (not01)[not01] , collapse = "', '" ) , "' in qMatrix contain elements that are not 0 or 1; this/these column(s) are ignored" ) , call. = FALSE )
+				qMatrix <- qMatrix[,colnames(qMatrix)[!colnames(qMatrix) %in% names (not01)[not01]],drop=FALSE]
 				}
 		}
 		# wenn nur eine Spalte wird diese als IDs angenommen
-		if ( !is.null ( qMatrix ) ) {
-				if ( ncol ( qMatrix ) %in% 1 ) {
-						warning ( "splitModels: qMatrix contains just one column; this is treated as item names" , call. = FALSE )
-						qMatrix$dim <- 1
-				}
+		if ( test_data_frame(qMatrix, ncols = 1) ) {
+		  warning ( "splitModels: qMatrix contains just one column; this is treated as item names" , call. = FALSE )
+			qMatrix$dim <- 1

```

Comment:
In general: Users may specify a Q matrix in `splitModels()` or `defineModel()` (or both). Both function are exported to the NAMESPACE. So it would be more efficient to capture all necessary checks in the `checkQmatrixConsistency()` function which will be called by `splitModels()` and `defineModel()`.